nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/TreeCommentManager.java

```diff
@@ -0,0 +1,490 @@
+            return commentModelToCreate;
+        });
+    }
+
+    @Override
+    public DocumentModel createLocatedComment(DocumentModel docModel, DocumentModel comment, String path) {
+        throw new UnsupportedOperationException(SERVICE_WITHOUT_IMPLEMENTATION_MESSAGE);
+    }
+
+    @Override
+    public DocumentModel createComment(DocumentModel docModel, String text) {
+        throw new UnsupportedOperationException(SERVICE_WITHOUT_IMPLEMENTATION_MESSAGE);
+    }
+
+    @Override
+    public DocumentModel createComment(DocumentModel docModel, String text, String author) {
+        throw new UnsupportedOperationException(SERVICE_WITHOUT_IMPLEMENTATION_MESSAGE);
+    }
+
+    @Override
+    public DocumentModel createComment(DocumentModel docModel, DocumentModel parent, DocumentModel child) {
+        throw new UnsupportedOperationException(SERVICE_WITHOUT_IMPLEMENTATION_MESSAGE);
+    }
+
+    @Override
+    public Comment updateComment(CoreSession s, String commentId, Comment comment) {
+        // Check the permissions
+        checkUpdateCommentPermissions(s, comment);
+
+        return CoreInstance.doPrivileged(s, session -> {
+            // Get the comment doc model
+            DocumentModel commentDocumentModel = getCommentDocumentModel(session, new IdRef(commentId));
+            // Initiate Modification Date if it is not done yet
+            if (comment.getModificationDate() == null) {
+                comment.setModificationDate(Instant.now());
+            }
+
+            Comments.commentToDocumentModel(comment, commentDocumentModel);
+            if (comment instanceof ExternalEntity) {
+                Comments.externalEntityToDocumentModel((ExternalEntity) comment, commentDocumentModel);
+            }
+            session.saveDocument(commentDocumentModel);
+            return Comments.newComment(commentDocumentModel);
+        });
+    }
+
+    @Override
+    public Comment updateExternalComment(CoreSession s, String entityId, Comment comment) {
+        // Check the permissions
+        checkUpdateCommentPermissions(s, comment);
+
+        return CoreInstance.doPrivileged(s, session -> {
+            // Get the external comment doc model
+            DocumentModel commentDocModel = getExternalCommentModel(session, entityId);
+            Comments.commentToDocumentModel(comment, commentDocModel);
+            if (comment instanceof ExternalEntity) {
+                Comments.externalEntityToDocumentModel((ExternalEntity) comment, commentDocModel);
+            }
+            session.saveDocument(commentDocModel);
+            return Comments.newComment(commentDocModel);
+        });
+    }
+
+    @Override
+    public void deleteExternalComment(CoreSession s, String entityId) {
+        DocumentModel commentDocModel = CoreInstance.doPrivileged(s, session -> {
+            return getExternalCommentModel(session, entityId);
```

Comment:
i thought to do it, but i should confess, i don't know if there is any side effects of calling `doProvilege` inside another () case of `update` for example [here](https://github.com/nuxeo/nuxeo/blob/1daebd3daee1ad241f98ce40f7233e961b6330ee/nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/TreeCommentManager.java#L257)

Reply:
There shouldn't be, in such cash, it just doing unneeded checks. But it's interesting to avoid them.

In the end, public methods (thus method from interface) could hold the logic to open privileged session (basically calls to `doPrivileged`) and protected methods shouldn't have this `doPrivileged` call, wdyt?